lib/bundler/cli/outdated.rb

```diff
@@ -30,6 +30,13 @@ def run
         Bundler.definition(:gems => gems, :sources => sources)
       end
 
+      patch_level = [:major, :minor, :patch].select {|v| options.keys.include?(v.to_s) }
+      Bundler.definition.create_gem_version_promoter([]).tap do |gvp|
+        gvp.level = patch_level.first || :major
+        gvp.strict = options[:strict]
+      end
+      Bundler.load
```

Comment:
We need to rebase to master and replace this section with `Bundler::CLI::Common.config_gem_version_promoter(Bundler.definition, options)`

This is going to cause a problem though which is a key part of this change: the meaning of the switches `--patch` / `--minor` / `--major` are now changing. Before they filtered the output, now they'll be influencing the resolution process.

There's only two ways here, though - we either CHANGE what they mean and rename the filtering ones to something like `--filter-patch` (which is breaking), or leave them as-is and introduce new flags (`--resolve-patch`, etc?) that are translated for GemVersionPromoter, which won't be breaking, but will be inconsistent with the flags for `bundle update`. 

My preference is to the former, those new flags weren't introduced too long ago, but I want @indirect's input on this.